src/app_service/service/keycardV2/queued_async_calls.nim

```diff
@@ -1,49 +1,31 @@
 proc onAsyncResponse(self: Service, response: string) {.slot.} =
   try:
     let responseObj = response.parseJson
+    let requestId = responseObj["requestId"].getInt()
+    if not self.requestMap.hasKey(requestId):
+      raise newException(Exception, "unexpected request id")
     if responseObj{"error"}.kind != JNull and responseObj{"error"}.getStr != "":
-      raise newException(CatchableError, responseObj{"error"}.getStr)
+      self.requestMap[requestId].callback(newJNull(), responseObj{"error"}.getStr)
```

Comment:
`self.requestMap.del(requestId)` is missing here

Reply:
perhaps a good place for `defer`

Reply:
yes, it's safe for defere here, updated, thanks